src/data_gradients/datasets/detection/voc_format_detection_dataset.py

```diff
@@ -154,25 +153,20 @@ def __init__(
         :param image_extensions:    List of image file extensions to load from.
         :param label_extensions:    List of label file extensions to load from.
         """
-        config_path = os.path.join(root_dir, config_path) if config_path is not None else None
-        self.image_label_tuples = ImageLabelFilesIterator(
-            images_dir=os.path.join(root_dir, images_subdir),
-            labels_dir=os.path.join(root_dir, labels_subdir),
+        super().__init__(
+            root_dir=root_dir,
+            images_subdir=images_subdir,
+            labels_subdir=labels_subdir,
             config_path=config_path,
+            verbose=verbose,
             image_extensions=image_extensions,
             label_extensions=label_extensions,
-            verbose=verbose,
         )
         self.class_names = class_names
 
-    def load_image(self, index: int) -> np.ndarray:
-        img_file, _ = self.image_label_tuples[index]
-        return load_image(path=img_file, channel_format=ImageChannelFormat.RGB)
-
-    def load_labels(self, index: int) -> np.ndarray:
-        _, label_path = self.image_label_tuples[index]
+    def load_labels(self, path: str) -> np.ndarray:
 
-        with open(label_path) as f:
+        with open(path) as f:
```

Comment:
I suggest to add `encoding="utf-8"` or wrap with try/catch to ensure we don't crash if for some XML would contain unicode symbols.

Reply:
[ddbc2a8](https://github.com/Deci-AI/data-gradients/pull/131/commits/ddbc2a8ced958f21f988333b3f873d459b8ffe84)